src/main/java/org/jenkinsci/maven/plugins/hpi/ValidateHpiMojo.java

```diff
@@ -41,17 +42,24 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     }
 
     private VersionNumber getDependencyCoreVersion(MavenArtifact artifact) throws IOException, MojoExecutionException {
-        Attributes mainAttributes = new JarFile(artifact.getFile()).getManifest().getMainAttributes();
-        Attributes.Name jName = new Attributes.Name("Jenkins-Version");
-        if (mainAttributes.containsKey(jName)) {
-            return new VersionNumber(mainAttributes.getValue(jName));
-        } else {
-            Attributes.Name hName = new Attributes.Name("Hudson-Version");
-            if (mainAttributes.containsKey(hName)) {
-                return new VersionNumber(mainAttributes.getValue(hName));
+        File file = artifact.getFile();
+        if (file.isFile()) {
+            Attributes mainAttributes = new JarFile(file).getManifest().getMainAttributes();
+            Attributes.Name jName = new Attributes.Name("Jenkins-Version");
+            if (mainAttributes.containsKey(jName)) {
+                return new VersionNumber(mainAttributes.getValue(jName));
             } else {
-                throw new MojoExecutionException("Could not find Jenkins Version in manifest for " + artifact);
+                Attributes.Name hName = new Attributes.Name("Hudson-Version");
+                if (mainAttributes.containsKey(hName)) {
+                    return new VersionNumber(mainAttributes.getValue(hName));
+                } else {
+                    throw new MojoExecutionException("Could not find Jenkins Version in manifest for " + artifact);
+                }
             }
+        } else {
+            getLog().warn("Skipping jenkins-core validation for " + artifact + " since we rely on sources and don't have a manifest. Use 'package' goal to get validation");
+            // Assume the version is the same
+            return new VersionNumber(findJenkinsVersion());
```

Comment:
IIRC this could be missing the core version check, as it is not really returning the dependency version but the current project core version. IMO, this should throw an exception pointing the user to run `mvn package` instead of `mvn compile` (or any goal before `package` in the maven lifecycle).

Proper validation runs on `mvn deploy` so releases are safe, but still... getting `mvn test` to not catch a Jenkins core version mismatch does not sound good.

Reply:
`mvn test` checks it through `InjectedTest`, see https://github.com/jenkinsci/maven-hpi-plugin/pull/191